hamilton/driver.py

```diff
@@ -1669,6 +1669,17 @@ def with_config(self, config: Dict[str, Any]) -> "Builder":
         self.config.update(config)
         return self
 
+    def with_local_modules(self) -> "Builder":
+        """Adds the local modules to the modules list.
```

Comment:
We should add more caveats here. 
E.g. If people use this, they could run into problems when using some of the adapters because this code will be under `__main__`.